File: 605 - Can Place Flowers/Solution.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        # Add 0s to the front and end of flowerbed list to handle edge cases
        f = [0] + flowerbed + [0]
        
        # Iterate over elements in f list, starting from the second element and ending at the second-to-last element
        for i in range(1, len(f) - 1):
            # If current element and the two elements before and after it are all 0s, plant a flower and decrement n
            if f[i - 1] == 0 and f[i] == 0 and f[i+1] == 0:
                f[i] = 1
                n-=1
        # Return True if n is less than or equal to 0, False otherwise
        return n <= 0

File: 605 - Can Place Flowers/Solution.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        // Counter for the number of flowers placed
        int count = 0;

        // Iterate through the flowerbed
        for (int i = 0; i < flowerbed.length; i++) {
            // Check if the current spot is empty, and if the spots before and after it (if
            // they exist) are also empty
            if (flowerbed[i] == 0 && (i == 0 || flowerbed[i - 1] == 0)
                    && (i == flowerbed.length - 1 || flowerbed[i + 1] == 0)) {
                // Place a flower in the current spot
                flowerbed[i] = 1;
                count++;
            }

            // Check if we've placed enough flowers
            if (count >= n) {
                return true;
            }
        }

        // Return false if we haven't placed enough flowers
        return false;
    }
}
